homeassistant/components/mazda/climate.py

```diff
@@ -0,0 +1,187 @@
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    ATTR_TEMPERATURE,
+    PRECISION_HALVES,
+    PRECISION_WHOLE,
+    UnitOfTemperature,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+from homeassistant.util.temperature import convert as convert_temperature
+
+from . import MazdaEntity
+from .const import DATA_CLIENT, DATA_COORDINATOR, DATA_REGION, DOMAIN
+
+PRESET_DEFROSTER_OFF = "Defroster Off"
+PRESET_DEFROSTER_FRONT = "Front Defroster"
+PRESET_DEFROSTER_REAR = "Rear Defroster"
+PRESET_DEFROSTER_FRONT_AND_REAR = "Front and Rear Defroster"
+
+
+def _front_defroster_enabled(preset_mode: str | None) -> bool:
+    return preset_mode in [
+        PRESET_DEFROSTER_FRONT_AND_REAR,
+        PRESET_DEFROSTER_FRONT,
+    ]
+
+
+def _rear_defroster_enabled(preset_mode: str | None) -> bool:
+    return preset_mode in [
+        PRESET_DEFROSTER_FRONT_AND_REAR,
+        PRESET_DEFROSTER_REAR,
+    ]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the climate platform."""
+    entry_data = hass.data[DOMAIN][config_entry.entry_id]
+    client = entry_data[DATA_CLIENT]
+    coordinator = entry_data[DATA_COORDINATOR]
+    region = entry_data[DATA_REGION]
+
+    async_add_entities(
+        MazdaClimateEntity(client, coordinator, index, region)
+        for index, data in enumerate(coordinator.data)
+        if data["isElectric"]
+    )
+
+
+class MazdaClimateEntity(MazdaEntity, ClimateEntity):
+    """Class for a Mazda climate entity."""
+
+    _attr_name = "Climate"
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.PRESET_MODE
+    )
+    _attr_hvac_modes = [HVACMode.HEAT_COOL, HVACMode.OFF]
+    _attr_preset_modes = [
+        PRESET_DEFROSTER_OFF,
+        PRESET_DEFROSTER_FRONT,
+        PRESET_DEFROSTER_REAR,
+        PRESET_DEFROSTER_FRONT_AND_REAR,
+    ]
+
+    def __init__(
+        self,
+        client: MazdaAPIClient,
+        coordinator: DataUpdateCoordinator,
+        index: int,
+        region: str,
+    ) -> None:
+        """Initialize Mazda climate entity."""
+        super().__init__(client, coordinator, index)
+
+        self.region = region
+        self._attr_unique_id = self.vin
+
+        if self.data["hvacSetting"]["temperatureUnit"] == "F":
+            self._attr_precision = PRECISION_WHOLE
+            self._attr_temperature_unit = UnitOfTemperature.FAHRENHEIT
+            self._attr_min_temp = 61.0
+            self._attr_max_temp = 83.0
+        else:
+            self._attr_precision = PRECISION_HALVES
+            self._attr_temperature_unit = UnitOfTemperature.CELSIUS
```

Comment:
Why do we need these two cases if the API always reports the temperature data in Celsius? The climate integration will convert between C and F for all the temperature attributes based on the climate entity `temperature_unit` attribute and the user globally configured temperature unit.

The idea is that the platform shouldn't need to bother with converting.